docs/proposals/kuratorcd/kurator-cd.md

```diff
@@ -0,0 +1,622 @@
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+### Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+In this section, we'll dive into the detailed API design for the Unified Continuous Delivery Feature.
+
+These APIs are designed to facilitate Kurator's integration with Flagger to enable the required functionality.
+
+Unlike Flagger, we may need to adjust Unified Continuous Delivery to reflect our new strategy and decisions.
+
+#### Unified Continuous Delivery API
+
+Kurator is designed to unify the installation of Flagger as a fleet plugin in a given single or multiple clusters.
+
+Then use the kurator application to distribute the Flagger configuration. Kurator's unified configuration distribution.
+
+Kurator puts the Continuous Delivery's api under the app's api, so that when Kurator deploys the app in the target cluster, it also deploys the corresponding Continuous Delivery policy.
+
+Here's the preliminary design for the Unified Continuous Delivery:
+
+```console
+// ApplicationSyncPolicy defines the configuration to sync an artifact.
+// Only oneof `kustomization` or `helm` can be specified to manage application sync.
+// ApplicationSyncPolicy distributes the Continuous Delivery configuration 
+// at the same time as the application deployment, if needed. 
+type ApplicationSyncPolicy struct {
+    // Name defines the name of the sync policy.
+    // If unspecified, a name of format `<application name>-<index>` will be generated.
+    // +optional
+    Name string `json:"name,omitempty"`
+
+    // Kustomization defines the configuration to calculate the desired state
+    // from a source using kustomize.
+    // +optional
+    Kustomization *Kustomization `json:"kustomization,omitempty"`
+    // HelmRelease defines the desired state of a Helm release.
+    // +optional
+    Helm *HelmRelease `json:"helm,omitempty"`
+
+    // Destination defines the destination for the artifact.
+    // If specified, it will override the destination specified at Application level.
+    // +optional
+    Destination *ApplicationDestination `json:"destination"`
+
+    // Delivery defines the Continuous Delivery Configurations to be used.
+    // If specified, a uniform Continuous Delivery policy is configured for this installed object.
+    // +optional
+    Delivery *DeliveryConfig `json:"deliveryPolicy"`
+}
+
+type DeliveryConfig struct {
+    // Autoscale defines minreplicas and maxreplicas of horizontal Pod Autoscaler.
+    // +optional
+    Autoscale *AutoscaleRef `json:"autoScale,omitempty"`
+
+    // Whether or not the testload is installed by the user. The default is false.
+    // If set it to ture, user need to install the testload himself.
```

Comment:
```suggestion
    // If set to true, user needs to install the testloader himself.
```

Reply:
I think the semantics is negative. 

We can make it: testLoader will be installed when this is true